octoprint_mrbeam/iobeam/hw_malfunction_handler.py

```diff
@@ -52,21 +78,42 @@ def report_hw_malfunction_from_plugin(self, malfunction_id, msg, payload=None):
             payload=payload,
         )
         dataset = {malfunction_id: data}
-
-        self.report_hw_malfunction(dataset, from_plugin=True)
+        if not self.error_already_reported(malfunction_id):
+            self.report_hw_malfunction(dataset, from_plugin=True)
+        else:
+            self._logger.debug("error already reported: %s", malfunction_id)
+
+    def error_already_reported(self, malfunction_id):
+        return self._messages_to_show.get(malfunction_id) is not None
+
+    @property
+    def hardware_malfunction(self):
+        if (
+            self._messages_to_show is None
+            or len(self._messages_to_show) == 0
```

Comment:
I think we should check for `not self._messages_to_show` before this, right?
And I think `self._messages_to_show == {}` would not be reached, as the length is already 0